web/src/features/fireWeather/components/graphs/PrecipGraph.tsx

```diff
@@ -147,6 +184,26 @@ const PrecipGraph: React.FunctionComponent<Props> = ({
       }
     )
 
+    const accumObservedPrecips: { date: Date; accumPrecip: number }[] = []
+    observedPrecips.forEach(({ date, value }) => {
+      const observed = { date, accumPrecip: NaN }
+      if (value != null) {
+        if (accumObservedPrecips.length === 0) {
+          observed.accumPrecip = value
+        } else {
+          observed.accumPrecip =
+            value + accumObservedPrecips[accumObservedPrecips.length - 1].accumPrecip
+        }
+        accumObservedPrecips.push(observed)
+      }
+    })
```

Comment:
Can you create a function out of this and use it for other weather values?

Reply:
sorry Kyubin, I'm not quite understanding what you're asking for here?

Reply:
Ah thought this can be refactored at the time of reviewing but now that I am looking at it again, it won't work very well :(